packages/opds-web-client/src/components/Book.tsx

```diff
@@ -1,15 +1,29 @@
 import * as React from "react";
 import CatalogLink from "./CatalogLink";
 import BookCover from "./BookCover";
+import BorrowButton from "./BorrowButton";
+import DownloadButton from "./DownloadButton";
 import { BookData } from "../interfaces";
+const download = require("downloadjs");
 
 export interface BookProps {
   book: BookData;
   collectionUrl?: string;
+  updateBook: (url: string) => Promise<BookData>;
+  fulfillBook: (url: string) => Promise<Blob>;
+  indirectFulfillBook: (url: string, type: string) => Promise<string>;
+  isSignedIn?: boolean;
+  epubReaderUrlTemplate?: (epubUrl: string) => string;
 }
 
-export default class Book extends React.Component<BookProps, void> {
+export default class Book<P extends BookProps> extends React.Component<P, void> {
+  constructor(props) {
+    super(props);
+    this.borrow = this.borrow.bind(this);
+  }
+
   render(): JSX.Element {
+    let summary = (this.props.book && this.props.book.summary && this.props.book.summary.replace(/<\/?[^>]+(>|$)/g, " ")) || "";
```

Comment:
This is checking that the summary is more than just HTML tags, right? Not actually removing HTML tags from the summary.

Reply:
No, it's actually removing the HTML tags, in order to provide a more compact view of the summary without any line breaks or styling that might be in the HTML.

Reply:
Then is `dangerouslySetInnerHTML={{__html: summary}}` just to be extra safe?

I don't have a strong opinion against stripping HTML out of the summary, it's just not what I was expecting.

Reply:
You're right, dangerouslySetInnerHTML is probably not necessary, I can remove that.

I'm stripping it because I noticed a bunch of books that started with 3 or 4 lines of like An Amazon Best Book of 2014, A Library Journal Best Book of 2014, etc and in list view that would end up as the entire summary. I wanted to fit as much as possible of the actual description in. You can click "More" to see the nicely formatted summary.

Reply:
Actually dangerouslySetInnerHTML is still needed to render HTML character entities which I'm not stripping out or changing.